junit5/src/main/java/sunstone/core/AwsCloudFormationCloudDeploymentManager.java

```diff
@@ -61,40 +63,38 @@ String deploy(String template, Map<String, String> parameters) {
         return stackName;
     }
 
-    @Override
-    public void undeploy(String id) {
+    public void undeploy(String stack) {
+        CloudFormationClient cfClient = stack2Client.get(stack);
         CloudFormationWaiter waiter = cfClient.waiter();
 
         DeleteStackRequest stackRequest = DeleteStackRequest.builder()
-                .stackName(id)
+                .stackName(stack)
                 .build();
 
         cfClient.deleteStack(stackRequest);
         DescribeStacksRequest stacksRequest = DescribeStacksRequest.builder()
-                .stackName(id)
+                .stackName(stack)
                 .build();
 
         WaiterResponse<DescribeStacksResponse> waiterResponse = waiter.waitUntilStackDeleteComplete(stacksRequest);
-        LOGGER.debug("Stack {} is deleted {}", id, waiterResponse.matched().response().orElse(null));
-    }
-
-    @Override
-    public void register(String id) {
-        stacks.add(id);
+        LOGGER.debug("Stack {} is deleted {}", stack, waiterResponse.matched().response().orElse(null));
+        stack2Client.remove(stack);
+        client2stacks.get(cfClient).remove(stack);
     }
 
     public void close() {
-        cfClient.close();
+        client2stacks.forEach((client, strings) -> client.close());
```

Comment:
???
 
So I will close the client for other stacks as well?

Reply:
Yes. This is done when the extension store is closed and won't be used in context of any other test class.

Reply:
Close method should be called once everything is undeployed and clients may be closed. Only clients that have been used for deployment operation are stored. We need to have them all since the client is bound to the region and an user may choose a region where the template should be deployed to.